dlp/sample.rb

```diff
@@ -0,0 +1,122 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def inspect_string string: nil, project_id: nil, max_findings: 0
+  # [START dlp_inspect_string]
+  # string       = "the text to inspect"
+  # project_id   = "Your Google Cloud project ID"
+  # max_findings = "maximum number of findings to report per request (0 = server maximum)"
+
+  require "google/cloud/dlp"
+
+  dlp = Google::Cloud::Dlp.new
+  inspect_config = {
+    # The types of information to match
+    info_types: [{name: "PERSON_NAME"}, {name: "US_STATE"}],
+
+    # Only return results above a likelihood threshold (0 for all)
+    min_likelihood: :POSSIBLE,
+
+    # Limit the number of findings (0 for no limit)
+    limits: { max_findings_per_request: max_findings },
+
+    # Whether to include the matching string in the response
+    include_quote: true
+  }
+
+  # The item to inspect
+  item_to_inspect = { value: string }
+
+  # Run request
+  parent = dlp.class.project_path project_id
+  response = dlp.inspect_content parent,
+    inspect_config: inspect_config,
+    item: item_to_inspect
+
+  # Print the results
+  response.result.findings.each do |finding|
+    puts "Quote:      #{finding.quote}"
+    puts "Info type:  #{finding.info_type.name}"
+    puts "Likelihood: #{finding.likelihood}"
+  end.empty? and begin
+    puts "No findings"
+  end
+  # [END dlp_inspect_string]
+end
+
+def inspect_file filename: nil, project_id: nil, max_findings: 0
+  # [START dlp_inspect_file]
+  # string       = "the text to inspect"
+  # project_id   = "Your Google Cloud project ID"
+  # max_findings = "maximum number of findings to report per request (0 = server maximum)"
+
+  require "google/cloud/dlp"
+
+  dlp = Google::Cloud::Dlp.new
+  inspect_config = {
+    # The types of information to match
+    info_types: [{name: "PERSON_NAME"}, {name: "PHONE_NUMBER"}],
+
+    # Only return results above a likelihood threshold (0 for all)
+    min_likelihood: :POSSIBLE,
+
+    # Limit the number of findings (0 for no limit)
+    limits: { max_findings_per_request: max_findings },
+
+    # Whether to include the matching string in the response
+    include_quote: true
+  }
+
+  # The item to inspect
+  file = File.open filename, "rb"
+  item = { byte_item: { type: :BYTES_TYPE_UNSPECIFIED, data: file.read } }
+
+  # Run request
+  parent = dlp.class.project_path project_id
```

Comment:
:slightly_frowning_face: @jbolinger, I'd like additional input here as this is not a great devex when trying to use `project_path` as it requires Ruby Metaprogramming. 

How about using the resource path: `"projects/#{project_id}"` instead?

Reply:
I wouldn't suggest using the `.class`. The following style is suggested in the generated documentation:

```ruby
Google::Cloud::Dlp::V2::DlpServiceClient.project_path project_id
```
It is quite long though...

Reply:
Thanks @jbolinger, 

I like that the client library feels version agnostic for the most part. I would push back on using `Google::Cloud::Dlp::V2::DlpServiceClient` as it's a simple helper [Class method](https://github.com/GoogleCloudPlatform/google-cloud-ruby-private/blob/dlpv2/google-cloud-dlp/lib/google/cloud/dlp/v2/dlp_service_client.rb#L197) and returns `projects/#{project_id}`. The main reason why `dlp.class.project_path` needs to be used is because it's a Class method and not an Instance method. 

Not sure if this is the **best** solution, change helper class methods to be instance methods instead. Is this something you can take a look at?

@bshaffer I wouldn't block samples on this issue, and would suggest resource path as a String literals instead.

Reply:
Yeah, I agree. It's not great but that's the current state.

Reply:
@jbolinger is this something you can take a look at fixing?

Reply:
@frankyn do you mean fixing the code generator or fixing this particular library for the v2 release?

I can't help with manual fixes to this library, but we could take a look at fixing it long term in the generator. That will take some time though. I don't know if it was done this way for a reason.

Reply:
FWIW in PHP we've implemented the `project_path` method as both a class and instance method.

Reply:
I meant the generator as it will impact other Auto generated libraries.

That would be most helpful moving forward and I understand it may take
time. Jacob may hopefully have more context as well.

Thanks @bshaffer, maybe that's a better option to fix this issue by defining the method as both a class and instance method.